Server/ReasnAPI/ReasnAPI/Services/EventService.cs

```diff
@@ -1,16 +1,276 @@
+            CreatedAt = eventToReturn.CreatedAt,
+            UpdatedAt = eventToReturn.UpdatedAt,
+            Slug = eventToReturn.Slug,
+            StatusId = eventToReturn.StatusId,
+            Tags = tags
+        };
+
+        return eventDto;
+    }
+
+    public IEnumerable<EventDto> GetEventsByFilter(Expression<Func<Event, bool>> filter)
+    {
+        var events = context.Events.Where(filter).ToList();
+        var eventDtos = new List<EventDto>();
+        foreach (var eventToReturn in events)
+        {
+            var tagIds = context.EventTags
+                .Where(r => r.EventId == eventToReturn.Id)
+                .Select(r => r.TagId)
+                .ToList();
+
+            var tags = context.Tags
+                .Where(t => tagIds.Contains(t.Id))
+                .Select(t => new TagDto { Name = t.Name })
+                .ToList();
+
+            var eventDto = new EventDto
+            {
+                Name = eventToReturn.Name,
+                AddressId = eventToReturn.AddressId,
+                Description = eventToReturn.Description,
+                OrganizerId = eventToReturn.OrganizerId,
+                StartAt = eventToReturn.StartAt,
+                EndAt = eventToReturn.EndAt,
+                CreatedAt = eventToReturn.CreatedAt,
+                UpdatedAt = eventToReturn.UpdatedAt,
+                Slug = eventToReturn.Slug,
+                StatusId = eventToReturn.StatusId,
+                Tags = tags
+            };
+
+            eventDtos.Add(eventDto);
+        }
+
+        return eventDtos;
+    }
+
+    public IEnumerable<EventDto> GetAllEvents()
+    {
+        var events = context.Events.ToList();
+        var eventDtos = new List<EventDto>();
+        foreach (var eventToReturn in events)
+        {
+            var tagIds = context.EventTags
+                .Where(r => r.EventId == eventToReturn.Id)
+                .Select(r => r.TagId)
+                .ToList();
+
+            var tags = context.Tags
+                .Where(t => tagIds.Contains(t.Id))
+                .Select(t => new TagDto { Name = t.Name })
+                .ToList();
+
+            var eventDto = new EventDto
+            {
+                Name = eventToReturn.Name,
+                AddressId = eventToReturn.AddressId,
+                Description = eventToReturn.Description,
+                OrganizerId = eventToReturn.OrganizerId,
+                StartAt = eventToReturn.StartAt,
+                EndAt = eventToReturn.EndAt,
+                CreatedAt = eventToReturn.CreatedAt,
+                UpdatedAt = eventToReturn.UpdatedAt,
+                Slug = eventToReturn.Slug,
+                StatusId = eventToReturn.StatusId,
+                Tags = tags
+            };
+
+            eventDtos.Add(eventDto);
+        }
+
+        return eventDtos;
+    }
+
+    private string CreateSlug(EventDto eventDto)
+    {
+        var slug = eventDto.Name.ToLower().Replace(" ", "-");

```

Comment:
How about other special characters? Needs to be HTML URL friendly, 
https://stackoverflow.com/questions/2920744/url-slugify-algorithm-in-c
https://github.com/fcingolani/Slugify